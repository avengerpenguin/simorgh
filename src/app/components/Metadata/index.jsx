import React from 'react';
import Helmet from 'react-helmet';
import { arrayOf, bool, string } from 'prop-types';

const Metadata = ({
  amp,
  articleAuthor,
  articleSection,
  canonicalLink,
  defaultImage,
  defaultImageAltText,
  description,
  lang,
  locale,
  metaTags,
  opengraphSiteName,
  timeFirstPublished,
  timeLastUpdated,
  title,
  twitterCreator,
  twitterSite,
  type,
}) => {
  const htmlAttributes = { lang };

  if (amp) {
    htmlAttributes.amp = amp;
  }

  return (
    <Helmet htmlAttributes={htmlAttributes}>
      <meta
        name="viewport"
        content="width=device-width, initial-scale=1, minimum-scale=1"
      />
      <title>{title}</title>
      <link rel="canonical" href={canonicalLink} />
      <meta name="article:author" content={articleAuthor} />
      <meta name="article:modified_time" content={timeLastUpdated} />
      <meta name="article:published_time" content={timeFirstPublished} />
      {articleSection ? (
        <meta name="article:section" content={articleSection} />
      ) : null}
      {metaTags.map(tag => (
        <meta name="article:tag" content={tag} key={tag} />
      ))}
      <meta name="og:description" content={description} />
      <meta name="og:image" content={defaultImage} />
      <meta name="og:image:alt" content={defaultImageAltText} />
      <meta name="og:locale" content={locale} />
      <meta name="og:site_name" content={opengraphSiteName} />
      <meta name="og:title" content={title} />
      <meta name="og:type" content={type} />
      <meta name="og:url" content={canonicalLink} />
      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:creator" content={twitterCreator} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:image:alt" content={defaultImageAltText} />
      <meta name="twitter:image:src" content={defaultImage} />
      <meta name="twitter:site" content={twitterSite} />
      <meta name="twitter:title" content={title} />
    </Helmet>
  );
};

Metadata.propTypes = {
  amp: bool.isRequired,
  articleAuthor: string.isRequired,
  articleSection: string,
  canonicalLink: string.isRequired,
  defaultImage: string.isRequired,
  defaultImageAltText: string.isRequired,
  description: string.isRequired,
  lang: string.isRequired,
  locale: string.isRequired,
  metaTags: arrayOf(string).isRequired,
  opengraphSiteName: string.isRequired,
  timeFirstPublished: string.isRequired,
  timeLastUpdated: string.isRequired,
  title: string.isRequired,
  twitterCreator: string.isRequired,
  twitterSite: string.isRequired,
  type: string.isRequired,
};

Metadata.defaultProps = {
  articleSection: null,
};

export default Metadata;
