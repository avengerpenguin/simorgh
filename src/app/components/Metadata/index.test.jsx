import React from 'react';
import Metadata from './index';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';

const metadataSnapshotTest = (
  testDescription,
  amp,
  articleAuthor,
  articleSection,
  canonicalLink,
  defaultImage,
  defaultImageAltText,
  description,
  lang,
  locale,
  metaTags,
  opengraphSiteName,
  timeLastUpdated,
  timeFirstPublished,
  title,
  twitterCreator,
  twitterSite,
) =>
  describe(testDescription, () => {
    const metadataProps = {
      amp,
      articleAuthor,
      articleSection,
      canonicalLink,
      defaultImage,
      defaultImageAltText,
      description,
      lang,
      locale,
      metaTags,
      opengraphSiteName,
      timeLastUpdated,
      timeFirstPublished,
      title,
      twitterCreator,
      twitterSite,
    };

    shouldShallowMatchSnapshot(
      'should render correctly',
      <Metadata {...metadataProps} />,
    );
  });

describe('Metadata', () => {
  metadataSnapshotTest(
    'News article',
    false,
    'BBC News',
    null,
    'https://www.bbc.com/news/articles/c0000000001o',
    'https://www.bbc.com/news/image.png',
    'BBC News',
    'This is a description',
    'en-GB',
    'en_GB',
    ['tagA', 'tagB'],
    'BBC News',
    1514811600000,
    1514811600000,
    'An article title',
    '@BBCNews',
    '@BBCNews',
  );

  metadataSnapshotTest(
    'News AMP article',
    true,
    'BBC News',
    null,
    'https://www.bbc.com/news/articles/amp/c0000000001o',
    'https://www.bbc.com/news/image.png',
    'BBC News',
    'This is a description',
    'en-GB',
    'en_GB',
    ['tagA', 'tagB'],
    'BBC News',
    1514811600000,
    1514811600000,
    'An article title',
    '@BBCNews',
    '@BBCNews',
  );

  metadataSnapshotTest(
    'Persian article',
    false,
    'BBC News فارسی',
    null,
    'https://www.bbc.com/persian/articles/c0000000028o',
    'https://www.bbc.com/persian/image.png',
    'BBC News فارسی',
    'This is a description',
    'fa',
    'fa',
    ['tagA', 'tagB'],
    'BBC News فارسی',
    1514811600000,
    1514811600000,
    'پهپادی که برایتان قهوه می‌آورد',
    '@bbcpersian',
    '@bbcpersian',
  );

  metadataSnapshotTest(
    'Persian AMP article',
    true,
    'BBC News فارسی',
    null,
    'https://www.bbc.com/persian/articles/amp/c0000000028o',
    'https://www.bbc.com/persian/image.png',
    'BBC News فارسی',
    'This is a description',
    'fa',
    'fa',
    ['tagA', 'tagB'],
    'BBC News فارسی',
    1514811600000,
    1514811600000,
    'پهپادی که برایتان قهوه می‌آورد',
    '@bbcpersian',
    '@bbcpersian',
  );

  metadataSnapshotTest(
    'articleSection is not null',
    false,
    'BBC News',
    'Politics',
    'https://www.bbc.com/news/articles/c0000000001o',
    'https://www.bbc.com/news/image.png',
    'BBC News',
    'This is a description',
    'en-GB',
    'en_GB',
    ['tagA', 'tagB'],
    'BBC News',
    1514811600000,
    1514811600000,
    'An article title',
    '@BBCNews',
    '@BBCNews',
  );
});
