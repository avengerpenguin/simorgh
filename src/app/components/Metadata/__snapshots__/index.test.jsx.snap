// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metadata News AMP article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "amp": true,
      "lang": "en-GB",
    }
  }
>
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    An article title
  </title>
  <link
    href="https://www.bbc.com/news/articles/amp/c0000000001o"
    rel="canonical"
  />
  <meta
    content="BBC News"
    name="article:author"
  />
  <meta
    content={1514811600000}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="og:image"
  />
  <meta
    content="BBC News"
    name="og:image:alt"
  />
  <meta
    content="en_GB"
    name="og:locale"
  />
  <meta
    content="BBC News"
    name="og:site_name"
  />
  <meta
    content="An article title"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/news/articles/amp/c0000000001o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@BBCNews"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@BBCNews"
    name="twitter:site"
  />
  <meta
    content="An article title"
    name="twitter:title"
  />
</HelmetWrapper>
`;

exports[`Metadata News article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en-GB",
    }
  }
>
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    An article title
  </title>
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    rel="canonical"
  />
  <meta
    content="BBC News"
    name="article:author"
  />
  <meta
    content={1514811600000}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="og:image"
  />
  <meta
    content="BBC News"
    name="og:image:alt"
  />
  <meta
    content="en_GB"
    name="og:locale"
  />
  <meta
    content="BBC News"
    name="og:site_name"
  />
  <meta
    content="An article title"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/news/articles/c0000000001o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@BBCNews"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@BBCNews"
    name="twitter:site"
  />
  <meta
    content="An article title"
    name="twitter:title"
  />
</HelmetWrapper>
`;

exports[`Metadata Persian AMP article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "amp": true,
      "lang": "fa",
    }
  }
>
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    پهپادی که برایتان قهوه می‌آورد
  </title>
  <link
    href="https://www.bbc.com/persian/articles/amp/c0000000028o"
    rel="canonical"
  />
  <meta
    content="BBC News فارسی"
    name="article:author"
  />
  <meta
    content={1514811600000}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="og:image"
  />
  <meta
    content="BBC News فارسی"
    name="og:image:alt"
  />
  <meta
    content="fa"
    name="og:locale"
  />
  <meta
    content="BBC News فارسی"
    name="og:site_name"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/persian/articles/amp/c0000000028o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@bbcpersian"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News فارسی"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@bbcpersian"
    name="twitter:site"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="twitter:title"
  />
</HelmetWrapper>
`;

exports[`Metadata Persian article should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "fa",
    }
  }
>
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    پهپادی که برایتان قهوه می‌آورد
  </title>
  <link
    href="https://www.bbc.com/persian/articles/c0000000028o"
    rel="canonical"
  />
  <meta
    content="BBC News فارسی"
    name="article:author"
  />
  <meta
    content={1514811600000}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="og:image"
  />
  <meta
    content="BBC News فارسی"
    name="og:image:alt"
  />
  <meta
    content="fa"
    name="og:locale"
  />
  <meta
    content="BBC News فارسی"
    name="og:site_name"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/persian/articles/c0000000028o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@bbcpersian"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News فارسی"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/persian/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@bbcpersian"
    name="twitter:site"
  />
  <meta
    content="پهپادی که برایتان قهوه می‌آورد"
    name="twitter:title"
  />
</HelmetWrapper>
`;

exports[`Metadata articleSection is not null should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
  htmlAttributes={
    Object {
      "lang": "en-GB",
    }
  }
>
  <meta
    content="width=device-width, initial-scale=1, minimum-scale=1"
    name="viewport"
  />
  <title>
    An article title
  </title>
  <link
    href="https://www.bbc.com/news/articles/c0000000001o"
    rel="canonical"
  />
  <meta
    content="BBC News"
    name="article:author"
  />
  <meta
    content={1514811600000}
    name="article:modified_time"
  />
  <meta
    content={1514811600000}
    name="article:published_time"
  />
  <meta
    content="Politics"
    name="article:section"
  />
  <meta
    content="tagA"
    name="article:tag"
  />
  <meta
    content="tagB"
    name="article:tag"
  />
  <meta
    content="This is a description"
    name="og:description"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="og:image"
  />
  <meta
    content="BBC News"
    name="og:image:alt"
  />
  <meta
    content="en_GB"
    name="og:locale"
  />
  <meta
    content="BBC News"
    name="og:site_name"
  />
  <meta
    content="An article title"
    name="og:title"
  />
  <meta
    name="og:type"
  />
  <meta
    content="https://www.bbc.com/news/articles/c0000000001o"
    name="og:url"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@BBCNews"
    name="twitter:creator"
  />
  <meta
    content="This is a description"
    name="twitter:description"
  />
  <meta
    content="BBC News"
    name="twitter:image:alt"
  />
  <meta
    content="https://www.bbc.com/news/image.png"
    name="twitter:image:src"
  />
  <meta
    content="@BBCNews"
    name="twitter:site"
  />
  <meta
    content="An article title"
    name="twitter:title"
  />
</HelmetWrapper>
`;
